### ISSUE TITLE
Error with Visual Studio 2017 15.9.1 - Cannot Use JSX

### ISSUE THREAD

**Issue by TheFanatr** [MicrosoftDocs/visualstudio-docs#1925]
Hello, I have followed this tutorial exactly and have been able to successfully compile the application; however, there is a persistent error in Visual Studio saying that the TypeScript compiler is complaining that I don't have the `--jsx` flag set and therefore cannot use JSX. This is indeed interfering with the editor's ability to provide proper IntelliSense and completion for JSX files. I have reattempted following the tutorial from scratch and observed the same result. The error is `TS17004`, and the message being shown is as follows.
```(TS) Cannot use JSX unless the '--jsx' flag is provided.```
I need to use Node-based server-side React for a project that I am working on, and, due to my familiarity with the product, was hoping I could use Visual Studio 2017. It would be very much appreciated if a resolution to issue this could be reached in the near future. Thanks.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: c4b0ea28-4a31-a3ea-fa5f-c32d5c72d4bd
* Version Independent ID: 31bd064b-51eb-5505-986e-2c7148784085
* Content: [Create a Node.js and React app - Visual Studio](https://docs.microsoft.com/en-us/visualstudio/javascript/tutorial-nodejs-with-react-and-jsx?view=vs-2017)
* Content Source: [docs/javascript/tutorial-nodejs-with-react-and-jsx.md](https://github.com/MicrosoftDocs/visualstudio-docs/blob/master/docs/javascript/tutorial-nodejs-with-react-and-jsx.md)
* Product: **visual-studio-dev15**
* GitHub Login: @Mikejo5000
* Microsoft Alias: **mikejo**

**Comment by Mikejo5000** [comment]
@TheFanatr Hi, what version of VS are you using? For example, 15.9.0 or similar?

**Comment by Mikejo5000** [comment]
Also, can you paste your dependencies here from the package.json file?

**Comment by Mikejo5000** [comment]
Right--15.9.1. Preview or no?

**Comment by Mikejo5000** [comment]
I can't repro this so far. Where do you see that error? I poked around a little. It seems like most instances of this error are in relation to the jsx entry in tsconfig.
 "jsx": "react",   Is that present? You might re-open project and re-try.

**Comment by TheFanatr** [comment]
I am currently on the latest stable release of Visual Studio 2017 15.9.1 using the latest version of both Node.js and NPM. I followed the instructions identically and copy-pasted all of the file contents from the tutorial. This is a fresh install of the Web Development with Node package straight out of the Visual Studio installer.

**Comment by TheFanatr** [comment]
I tried making the project twice, once adding the dependencies though the GUI option, always choosing the latest version, then once that one didn't work even after I tried playing with the settings, I deleted the entire repo and tried again with the exact dependency configuration in the tutorial to no avail.

**Comment by Mikejo5000** [comment]
@billti Bill, do you know what might cause this error?

**Comment by Mikejo5000** [comment]
@TheFanatr Just updated VS and retested, still no repro on this. You said that it compiles fine. Does it also run fine via F5? Where do you see the TS17004 error?

This GitHub issue seems to describe this issue for TypeScript in VS:
https://github.com/Microsoft/TypeScript/issues/7394
Based on this, I'd ask:
- How did you add the tsconfig
- What is Build Action set to?
There's also a link to a stack post that might have some useful info. Other than that, I'd need your project to test on.

**Comment by Mikejo5000** [comment]
@armanio123 also for comment

**Comment by TheFanatr** [comment]
For future reference in case the issue reappears, I'll answer the questions, but after the update to Visual Studio 15.9.2 and a computer restart, this issue seems to have been fixed. When it was broken, the project would run fine via any valid way to run it, but two instances of the aforementioned error would persist in the error list no matter what, one for the return statement in the `render` method on the `Hello` component and one where the component was referenced in the call to `ReactDOM.render` at the bottom of the page; this is, of course, using the JSX code provided by the tutorial. I added the "tsconfig.json" file via the prompt shown via `Add > New Item` on the project node's context menu. The build action is set to `Content` now, but I don't know what it was set to before. I think this issue can be closed now.
